staking/contracts/contracts.py

```diff
@@ -36,38 +38,47 @@ def send_asset(
     recipient: abi.Account
 ) -> Expr:
     return Seq(
+        
+        # Store amount we actually need to dispense.
+        (dispensed_amount := ScratchVar()).store(amount.get()),
```

Comment:
controversial nit: consider having a new function `send_all_asset(asset, recipient)` in addition to this one and have separate code paths for the case that a user wants to have some vs. ALL of their amount withdrawn. It would reduces the chance of someone withdrawing everything because of a fat finger.

Reply:
So in the case that an amount is requested > amount available, we would fail instead of withdrawing all.